src/rules/padding-before-describe-blocks.ts

```diff
@@ -1,48 +1,14 @@
-/**
- * @fileoverview Enforces single line padding around describe blocks
- * @author Dan Green-Leipciger
- */
-'use strict';
-
-import { padBefore } from '../utils';
-//------------------------------------------------------------------------------
-// Rule Definition
-//------------------------------------------------------------------------------
-
-const beforeMessage =
-  'You need a newline or comment before a describe block when it comes after another expression';
-
-const isDescribe = node =>
-  node.expression &&
-  node.expression.callee &&
-  node.expression.callee.name === 'describe';
+import padding from './padding';
 
 export default {
-  meta: {
-    docs: {
-      description: 'Enforces at least a line of padding before describe blocks',
-      category: 'Fill me in',
-      recommended: false,
-    },
-    fixable: 'whitespace', // or "code" or "whitespace"
-    schema: [
-      // fill in your schema
-    ],
-  },
-  beforeMessage,
+  ...padding,
   create(context) {
-    const filePath = context && context.getFilename();
-    const isTest =
-      filePath.includes('test') ||
-      filePath.includes('spec') ||
-      filePath.includes('<input>');
-    return {
-      ExpressionStatement(node) {
-        if (!isTest || !isDescribe(node)) {
-          return;
-        }
-        padBefore({ context, node, beforeMessage });
-      },
-    };
+    const ctx = Object.create(context, {
```

Comment:
Here we have to use `Object.create` to alter the `RuleContext` and specify the options that implement the describe rules. The `RuleContext` we receive is frozen.

Reply:
This is how all of the singular rules would be implemented now to preserve the old rules api. The generic padding rule would also be available directly in the eslintrc

Reply:
(we should probably freeze the new `RuleContext` again before we send it along)